### PR TITLE
Migrate Drawer to TS

### PR DESCRIPTION
**Pull request by ethanshar**
Also fixed minor issue with drawer toggle invocation. (What I sent you in slack)

### CODE REVIEW

File: generatedTypes/components/drawer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+     */
+    onWillFullSwipeRight?: Function;
+    /**
+     * Haptic trigger function to use onToggleSwipeLeft
+     */
+    leftToggleHapticTrigger?: Function;
+    /**
+     * Style
+     */
+    style: ViewStyle;
```
Comment (Inbal-Tish) [inline generatedTypes/components/drawer/index.d.ts:94]: Style is not optional?

File: generatedTypes/components/drawer/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+}
+/**
+ * @description: Drawer Component
+ * @important: If your app works with RNN, your screen must be wrapped
+ * with gestureHandlerRootHOC from 'react-native-gesture-handler'. see
+ * @importantLink: https://kmagiera.github.io/react-native-gesture-handler/docs/getting-started.html#with-wix-react-native-navigation-https-githubcom-wix-react-native-navigation
+ */
+declare class NewDrawer extends PureComponent<DrawerProps> {
```
Comment (Inbal-Tish) [inline generatedTypes/components/drawer/index.d.ts:102]: This file should be deleted, no?

File: src/components/drawer/index.tsx [hunk 3]
```diff
@@ -1,217 +1,221 @@
...
+   */
+  onWillFullSwipeRight?: Function;
+  /**
+   * Haptic trigger function to use onToggleSwipeLeft
+   */
+  leftToggleHapticTrigger?: Function;
+  /**
+   * Style
+   */
+  style: ViewStyle;
```
Comment (Inbal-Tish) [inline src/components/drawer/index.tsx:107]: It here...

File: src/components/drawer/index.tsx [hunk 4]
```diff
@@ -244,22 +248,22 @@ class NewDrawer extends PureBaseComponent {
...
+    return this.renderActions(rightItems, progress/* , dragX */);
   };
 
-  renderActions(items, progress, dragX) {
+  private renderActions(items: ItemProps[] | undefined, progress: Animated.Value/* , dragX: Animated.Value */) {
     if (items) {
       return (
-        <View animated row style={{transform: [{translateX: this.leftActionX}]}}>
-          {_.map(items, (item, index) => {
+        <View animated row style={{transform: [{translateX: this.leftActionX}]} as any as ViewStyle}>
```
Comment (Inbal-Tish) [inline src/components/drawer/index.tsx:265]: Why you need ' as any as ViewStyle' here?

Comment (ethanshar) [reply src/components/drawer/index.tsx:265]: TBH, not sure why it's even giving me an error. 
This is a trick I learned from the guys from Kiev. 
First you cast to any and from there you can cast to anything you want. 

Not a fan of this solution. another option is to explicitly `ts-ignore` this line and revisit in the future. 
WDYT?

File: src/components/drawer/index.tsx (hunk 3 above)
Comment (ethanshar) [reply src/components/drawer/index.tsx]: Fixed

File: generatedTypes/components/drawer/index.d.ts (hunk 2 above)
Comment (ethanshar) [reply generatedTypes/components/drawer/index.d.ts]: Fixed

File: generatedTypes/components/drawer/index.d.ts (hunk 1 above)
Comment (ethanshar) [reply generatedTypes/components/drawer/index.d.ts]: Indeed. Fixed

File: src/components/drawer/index.tsx (hunk 4 above)
Comment (Inbal-Tish) [reply src/components/drawer/index.tsx:264]: I would ts-ignore it, the casting seems redundant and expensive

Comment (ethanshar) [reply src/components/drawer/index.tsx]: Done

**Pull request merged**